src/discrete-layer/components/layer-details/field-value-presentors/json.value-presentor.tsx

```diff
@@ -1,87 +1,235 @@
+        if (fileType === ZIP_EXTENSION) {
+          void shp(shapeArrayBuffer)
+            .then((data) => {
+              // Extract polygon from geoJson featureCollection
+              const featuresArr = (data as FeatureCollectionWithFilename).features;
+              const shapeFileValidation = isShapeFileValid(featuresArr);
+
+              if(shapeFileValidation instanceof Error) {
+                return reject(shapeFileValidation);
+              }
 
+              const geometryPolygon = featuresArr[0].geometry;
+              return resolve(geometryPolygon);
+            })
+            .catch(() => {
+              return reject(shapeFileGenericError);
+            });
+        } else {
+          // Probably is shape file.
+          const geometryArr = parseShp(shapeArrayBuffer);
+          const shapeFileValidation = isShapeFileValid(geometryArr);
+              
+          if(shapeFileValidation instanceof Error) {
+            return reject(shapeFileValidation);
+          }
+
+          const geometryPolygon = geometryArr[0];
+          return resolve(geometryPolygon);
+        }
+      } catch (e) {
+        return reject(shapeFileGenericError);
+      }
+    });
+  };
+
+  const renderLoadShapeFileBtn = (): JSX.Element => {
+    const btnText = intl.formatMessage({
+      id: 'button-field-text.footprint.load-from-shapeFile',
+    });
+
+    return (
+      <Box className="loadFromShapeBtn">
+        <Button
+          outlined
+          type="button"
+          onClick={(): void => {
+            importShapeFileFromClient((ev, fileType) => {
+              const shpFile = (ev.target?.result as unknown) as ArrayBuffer;
+              void proccessShapeFile(shpFile, fileType)
+                .then((geometryPolygon) => {
+                  setJsonValue(JSON.stringify(geometryPolygon));
+                  removeStatusErrors();
+                })
+                .catch((e) => {
+                  const errorTranslation = intl.formatMessage(
+                    { id: (e as Error).message },
+                    {
+                      fieldName: emphasizeByHTML(
+                        `${intl.formatMessage({ id: fieldInfo.label })}`
+                      ),
+                    }
+                  );
 
-  if (formik === undefined || mode === Mode.VIEW || (mode === Mode.EDIT && fieldInfo.isManuallyEditable !== true)) {
+                  formik?.setStatus({
+                    errors: {
+                      ...currentErrors,
+                      [fieldInfo.fieldName as string]: [errorTranslation],
+                    },
+                  });
+                  setJsonValue('{}');
+                })
+                .finally(() => {
+                  // Focus input after loading shape file, validations occurs on blur.
+                  fieldRef.current?.focus();
+                });
+            });
+          }}
+        >
+          {btnText}
+        </Button>
+      </Box>
+    );
+  };
+
+  if (
```

Comment:
SURE?

Reply:
If you mean the auto-focus, I did it to "enforce" validations for the imported json.. Do you think this is unnecessary?

Reply:
Yeah its ok